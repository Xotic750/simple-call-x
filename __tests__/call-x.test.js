import call from '../src/call-x';

describe('testSubject', function() {
  it('is a function', function() {
    expect.assertions(1);
    expect(typeof call).toBe('function');
  });

  it('should throw when target not a function', function() {
    expect.assertions(6);
    expect(function() {
      call();
    }).toThrowErrorMatchingSnapshot();
    expect(function() {
      call(undefined);
    }).toThrowErrorMatchingSnapshot();
    expect(function() {
      call(null);
    }).toThrowErrorMatchingSnapshot();
    expect(function() {
      call(1);
    }).toThrowErrorMatchingSnapshot();
    expect(function() {
      call(true);
    }).toThrowErrorMatchingSnapshot();
    expect(function() {
      call('');
    }).toThrowErrorMatchingSnapshot();
  });

  it('context', function() {
    expect.assertions(1);
    const context = {};
    expect(
      call(function() {
        /* eslint-disable-next-line babel/no-invalid-this */
        return this;
      }, context),
    ).toBe(context);
  });

  it('args', function() {
    expect.assertions(1);
    const context = {};
    expect(
      call(
        function(a, b, c) {
          /* eslint-disable-next-line babel/no-invalid-this */
          return {context: this, a, b, c};
        },
        context,
        [1, 2, 3],
      ),
    ).toMatchSnapshot();
  });

  it('args string', function() {
    expect.assertions(1);
    const context = {};
    expect(
      call(
        function(a, b, c) {
          /* eslint-disable-next-line babel/no-invalid-this */
          return {context: this, a, b, c};
        },
        context,
        '123',
      ),
    ).toMatchSnapshot();
  });

  it('args other', function() {
    expect.assertions(1);
    const context = {};
    expect(
      call(
        function(a, b, c) {
          /* eslint-disable-next-line babel/no-invalid-this */
          return {context: this, a, b, c};
        },
        context,
        true,
      ),
    ).toMatchSnapshot();
  });
});
