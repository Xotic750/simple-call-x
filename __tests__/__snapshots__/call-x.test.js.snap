// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testSubject args 1`] = `
Object {
  "a": 1,
  "b": 2,
  "c": 3,
  "context": Object {},
}
`;

exports[`testSubject args other 1`] = `
Object {
  "a": undefined,
  "b": undefined,
  "c": undefined,
  "context": Object {},
}
`;

exports[`testSubject args string 1`] = `
Object {
  "a": "1",
  "b": "2",
  "c": "3",
  "context": Object {},
}
`;

exports[`testSubject should throw when target not a function 1`] = `"undefined is not a function"`;

exports[`testSubject should throw when target not a function 2`] = `"undefined is not a function"`;

exports[`testSubject should throw when target not a function 3`] = `"null is not a function"`;

exports[`testSubject should throw when target not a function 4`] = `"1 is not a function"`;

exports[`testSubject should throw when target not a function 5`] = `"true is not a function"`;

exports[`testSubject should throw when target not a function 6`] = `" is not a function"`;
